test/tests/postprocessors/residual/gold/current_residual_out.csv

```diff
@@ -0,0 +1,4 @@
+time,test_pp
+0.001,3.0822070014845
+0.002,1.3738200122113e-06
+0.003,1.3515558779779e-11
```

Comment:
diffing a residual seems pretty sensitive to future changes in the solver or if someday we have a recipe that runs every test with `-vec_type cuda -mat_type aijcusparse` etc. It seems like there is likely a better way to ensure that your new execution flag is working, which is the core of this PR. Sadly I cannot think of a better way to test your changes to the `Residual` class, which is less central to your PR